### PR TITLE
Fix sorting in limit expression node context

### LINKED ISSUE #33903: Improve sorting in limit expression node context
**Description:**
limit-clause

As depicted by the above capture, the expressions after `limit` keyword are not sorted correctly. According to the spec the expressions that are allowed in the said context should have a subtype of `int` . Those should be given the highest precedence. 

**Steps to reproduce:**
```ballerina
function findToptenCountriesByAthletes() returns error? {
    int zlimit = 10;
    string alimit = "";
    map<int> athletesByCountry = check findNumOfAthletesBycountry();
    string[] listResult = from [string,int] entry in athletesByCountry.entries() 
                     order by entry[1] 
                     limit  <cursor> select entry[0];
    
    io:println("Top 10 countries by number of Athletes: ");
    foreach string item in listResult {
              io:println("\t", item);
    }
}
```

**Affected Versions:**
slbeta4-rc2

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
This PR improves the sorting in limit expression node context.

Fixes #33903 

## Samples
Screenshot 2022-08-02 at 17 43 17

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import ballerina/module1;
+
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal:2]: Shall we include a `int`  constant and a function which returns int as well to ensure they are sorted properly as well

**Review by malinthar — COMMENTED** [review]
LGTM!

File: language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal:2]: +1

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LimitClauseNodeContext.java [hunk 2]
```diff
@@ -74,4 +77,18 @@ public boolean onPreValidation(BallerinaCompletionContext context, LimitClauseNo
...
     }
+
+    @Override
+    public void sort(BallerinaCompletionContext context, LimitClauseNode node, List<LSCompletionItem> completionItems) {
+        Optional<SemanticModel> semanticModel = context.currentSemanticModel();
+        if (semanticModel.isEmpty()) {
+            super.sort(context, node, completionItems);
+            return;
+        }
+        TypeSymbol typeSymbol = semanticModel.get().types().INT;
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LimitClauseNodeContext.java:88]: We can use more descriptive variable name like `intTypeSymbol` WDYT?

File: language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal (hunk 1 above)
Comment (gayalkuruppu) [reply language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal:2]: +1

Comment (mindula) [reply language-server/modules/langserver-core/src/test/resources/completion/query_expression/source/query_expr_ctx_limit_clause_source4.bal:2]: Will add it.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LimitClauseNodeContext.java (hunk 2 above)
Comment (mindula) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LimitClauseNodeContext.java:88]: +1

**Review by gayalkuruppu — APPROVED** [review]
LGTM

**Review by IMS94 — APPROVED** [review]

**Pull request merged**